### DISCUSSION TITLE
Web Extension specific `extensionDependencies`

### DISCUSSION THREAD

**Discussion by nturinski** [microsoft/vscode-discussions#462] [Extension Development QnA] [answered]
Is there any property/field to specify an extensionDependency is _only_ for web or desktop? One of our extension dependencies is not web ready, but we have an alternative path to mitigate not being able to install this dependency.

Alternatively, is there a way to upload two separate `package.jsons` to the marketplace? I know that is not recommended practice, but the only difference would be `extensionDependencies` array. All commands will be enabled/disabled via the `enablement` clauses.

**Comment by alefragnani** [comment, marked as answer]
Hi @nturinski ,

Right now, there is no way to distinguish between web or desktop `devDependencies`. Once you add it, it will be available for both.

I'm not a Webpack power user but, based on the [Webpack configuration](https://code.visualstudio.com/api/extension-guides/web-extensions#webpack-configuration) `resolve` tip for libs that aren't available on the web, I wonder if you could use similar approach (or other Webpack feature) to package your extension with the desired libs for each platform.

About multiple `package.json` for a single extension, I remember of a discussion (I guess it was on the VS Code Development Community Slack channel) about similar request, and if I'm right, it is not possible. But I guess this could be accomplished, if `vsce` CLI is updated to allow define _alternative_ `package.json`, similar how it does for `.vscodeignore` file.

Hope this helps
